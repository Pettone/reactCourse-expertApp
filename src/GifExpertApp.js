import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    /* Si se desea agregar más elementos, no se podría. No se puede mutar un arreglo así en React.
      En cambio debe usarse el Hook useState*/
    //const categories = ['One Punch', 'Samurai X','Dragon Ball'];

    const [ categories, setCategories ] = useState(['Guardians Of The Galaxy']);

    return(
      <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories = { setCategories } />
        <hr />
      
        <ol>
          {
            categories.map( category => (
              <GifGrid 
                  key = { category }
                  category = { category }
              />
            ))
          }
        </ol>
      </>
    );
  }
