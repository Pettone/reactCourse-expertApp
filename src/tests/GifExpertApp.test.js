import React from 'react';
import { render } from '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifExpertApp } from '../GifExpertApp';

describe('Testing GifExpertApp component', () => { 

    test('Must to show itself correctly', () => { 

        const wrapper = shallow( <GifExpertApp /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('Must to show a category list', () => { 

        const categories = ['Dragonball','Pokemon'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories}/> );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});