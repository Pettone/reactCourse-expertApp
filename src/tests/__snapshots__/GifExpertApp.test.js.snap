// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp component Must to show a category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragonball"
      key="Dragonball"
    />
    <GifGrid
      category="Pokemon"
      key="Pokemon"
    />
  </ol>
</Fragment>
`;

exports[`Testing GifExpertApp component Must to show itself correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
